src/Php/PhpVersionFactoryFactory.php

```diff
@@ -43,7 +49,20 @@ public function create(): PhpVersionFactory
 			}
 		}
 
-		return new PhpVersionFactory($this->versionId, $composerPhpVersion);
+		$versionId = null;
+
+		if (is_int($this->phpVersion)) {
+			$versionId = $this->phpVersion;
+		}
+
+		if ($this->bleedingEdge && is_array($this->phpVersion)) {
+			if (!array_key_exists('min', $this->phpVersion)) {
```

Comment:
This check doesn't have to be here, it's already enforced by the constructor array shape.